tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPConnectSaslTest.java

```diff
@@ -151,6 +164,34 @@ public void testConnectsWithPlain() throws Exception {
       assertArrayEquals(expectedPlainInitialResponse(USER, PASSWD), authenticator.getInitialResponse());
    }
 
+   @Test(timeout = 200000)
+   public void testConnectsWithSCRAM() throws Exception {
+      CountDownLatch serverConnectionOpen = new CountDownLatch(1);
+      SCRAMTestAuthenticator authenticator = new SCRAMTestAuthenticator(SCRAM.SHA512);
```

Comment:
If the server does not choose the offered mechanism the test simply fails as it requires SASL-SCRAM in the TestConnector

Reply:
I realise that, it is part of my point. Since it is the only mechanism offered by the server, then the connecting broker clearly had to pick that mechanism (or fail if it doesnt support it at all). Offering other mechs (as many of the other tests in the class do) then also verifies the actual selection behaviour, i.e that it will prefer selecting SCRAM 512 when it is offered, rather than only doing so because it had no other choice. That way if someone accidentally breaks that logic later the test can at least point it out.